### PR TITLE
NCL-4715 Use single DA invocation 

### PR DESCRIPTION
**Pull request by rnc**
This also refactors the packages and changes the cache to use a single object.

### CODE REVIEW

**Comment by rnc** [comment]
**Note** - this is not mandatory for the MVP so we can delay merging if we want.

**Comment by geoand** [comment]
I took a quick look (only at the actual behavior changes - didn't really bother with the package changes and such) and I really like it.
I'll have to give it a go locally tomorrow but if it works for the projects we test it against and if you are confident with it, I think we should merge it

File: analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java [hunk 1]
```diff
@@ -55,37 +55,66 @@ public void perform() {
...
+            if (projectsToAlign.isEmpty()) {
+
+                Collection<ProjectVersionRef> allDeps = cache.getDependencies().values().stream().flatMap(Collection::stream)
+                        .collect(Collectors.toList());
+
+                final AlignmentService alignmentService = AlignmentServiceFactory
+                        .getAlignmentService(cache.getDependencies().keySet());
+                final AlignmentService.Response alignmentResponse = alignmentService.align(
+                        new AlignmentService.Request(
+                                cache.getGAV().stream().findFirst().get(),
```
Comment (geoand) [inline analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java]: Is this safe?

**Comment by rnc** [comment]
@TomasHofman  Do you want to take a look at this as well?

**Comment by TomasHofman** [comment]
Checking..

**Comment by TomasHofman** [comment]
+1, I like it too. It makes the process lot clearer.

File: analyzer/src/functTest/java/org/jboss/gm/analyzer/alignment/GrpcLikeLayoutFunctionalTest.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import org.jboss.gm.common.utils.FileUtils;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.RestoreSystemProperties;
+import org.junit.contrib.java.lang.system.SystemOutRule;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.TestRule;
+
+public class GrpcLikeLayoutFunctionalTest extends AbstractWiremockTest {
```
Comment (geoand) [inline analyzer/src/functTest/java/org/jboss/gm/analyzer/alignment/GrpcLikeLayoutFunctionalTest.java]: :100:

**Pull request merged**